File: task3_file_security/encryption.py
import base64
from logger import log_action


def encrypt_file(filename):
    try:
        with open(filename, "rb") as file:
            data = file.read()

        encrypted = base64.b64encode(data)

        with open(filename, "wb") as file:
            file.write(encrypted)

        print("File encrypted successfully.")
        log_action(f"Encrypted file: {filename}")

    except FileNotFoundError:
        print("File not found.")


def decrypt_file(filename):
    try:
        with open(filename, "rb") as file:
            data = file.read()

        decrypted = base64.b64decode(data)

        with open(filename, "wb") as file:
            file.write(decrypted)

        print("File decrypted successfully.")
        log_action(f"Decrypted file: {filename}")

    except FileNotFoundError:
        print("File not found.")

    except Exception:
        print("Unable to decrypt file.")

File: task3_file_security/encryption.c
#include <stdio.h>
#include <stdlib.h>
#include "encryption.h"
#include "logger.h"

#define KEY 5


void encryptFile()
{
    char filename[100];

    printf("Enter filename to encrypt: ");
    scanf("%99s", filename);


    FILE *file = fopen(filename, "r");

    if (file == NULL)
    {
        printf("File not found.\n");
        return;
    }


    FILE *temp = fopen("encrypted.tmp", "w");


    char ch;

    while ((ch = fgetc(file)) != EOF)
    {
        fputc(ch + KEY, temp);
    }


    fclose(file);
    fclose(temp);


    remove(filename);
    rename("encrypted.tmp", filename);


    logAction("ENCRYPT", filename);

    printf("File encrypted successfully.\n");
}



void decryptFile()
{
    char filename[100];

    printf("Enter filename to decrypt: ");
    scanf("%99s", filename);


    FILE *file = fopen(filename, "r");

    if (file == NULL)
    {
        printf("File not found.\n");
        return;
    }


    FILE *temp = fopen("decrypted.tmp", "w");


    char ch;

    while ((ch = fgetc(file)) != EOF)
    {
        fputc(ch - KEY, temp);
    }


    fclose(file);
    fclose(temp);


    remove(filename);
    rename("decrypted.tmp", filename);


    logAction("DECRYPT", filename);

    printf("File decrypted successfully.\n");
}
